pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -282,7 +280,21 @@ func (e *eksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 			}
 		}
 
-		if cluster.Status.ServiceAccountToken == "" || cluster.Status.APIEndpoint == "" {
+		if cluster.Status.APIEndpoint == "" {
+			return e.getCAAndAPIEndpoint(cluster)
```

Comment:
does this function record something on the cluster? If so, the name should be changed or record should happen outside of it. I know it's not part of this PR but I usually do not expect mutations from get functions. Disclaimer: totally possible I wrote it 😅

Reply:
Yes, that is a bad name.